gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -875,6 +878,59 @@ public boolean delete(Path hadoopPath, boolean recursive) throws IOException {
     return status.toArray(new FileStatus[0]);
   }
 
+  class GCSFilesListIterator implements RemoteIterator<LocatedFileStatus> {
```

Comment:
May you inline this class? something like: https://github.com/apache/hadoop/blob/b6c06c4b76ca5c206639e2ddea4f3bef393dbb11/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java#L2178

Reply:
Would it be possible to use `FileStatus` instead of `LocatedFileStatus`

```
class GCSFilesListIterato<T extends FileStatus>r implements RemoteIterator<T>
```

Reply:
Would it be more accurate to use `GCSDirectoryListingIterator` ?